models/google_play__overview_report.sql

```diff
@@ -0,0 +1,187 @@
+    from store_performance_rollup
+), 
+
+overview_join as (
+
+    select 
+        -- these 2 columns are the grain of this model
+        coalesce(install_metrics.date_day, ratings.date_day, store_performance_metrics.date_day, crashes.date_day) as date_day,
+        coalesce(install_metrics.package_name, ratings.package_name, store_performance_metrics.package_name, crashes.package_name) as package_name,
+
+        -- metrics based on unique devices + users
+        coalesce(install_metrics.active_device_installs, 0) as active_device_installs,
+        coalesce(install_metrics.daily_device_installs, 0) as daily_device_installs,
+        coalesce(install_metrics.daily_device_uninstalls, 0) as daily_device_uninstalls,
+        coalesce(install_metrics.daily_device_upgrades, 0) as daily_device_upgrades,
+        coalesce(install_metrics.daily_user_installs, 0) as daily_user_installs,
+        coalesce(install_metrics.daily_user_uninstalls, 0) as daily_user_uninstalls,
+        coalesce(store_performance_metrics.store_listing_acquisitions, 0) as store_listing_acquisitions,
+        coalesce(store_performance_metrics.store_listing_visitors, 0) as store_listing_visitors,
+        store_performance_metrics.store_listing_conversion_rate, -- not coalescing if there aren't any visitors 
+
+        -- metrics based on events. a user or device can have multiple installs in one day
+        coalesce(crashes.daily_crashes, 0) as daily_crashes,
+        coalesce(crashes.daily_anrs, 0) as daily_anrs,
+        coalesce(install_metrics.install_events, 0) as install_events,
+        coalesce(install_metrics.uninstall_events, 0) as uninstall_events,
+        coalesce(install_metrics.update_events, 0) as update_events,    
+
+        -- all of the following fields (except daily_average_rating) are rolling metrics that we'll use window functions to backfill instead of coalescing
+        install_metrics.total_unique_user_installs,
+        install_metrics.total_device_installs,
+        install_metrics.total_device_uninstalls,
+        ratings.daily_average_rating, -- this one actually isn't rolling but we won't coalesce days with no reviews to 0 rating. todo: move
+        ratings.rolling_total_average_rating,
+        store_performance_metrics.total_store_acquisitions,
+        store_performance_metrics.total_store_visitors
+
+    from install_metrics
+    full outer join ratings
+        on install_metrics.date_day = ratings.date_day
+        and install_metrics.package_name = ratings.package_name
+    full outer join store_performance_metrics
+        on store_performance_metrics.date_day = install_metrics.date_day
+        and store_performance_metrics.package_name = install_metrics.package_name
+    full outer join crashes
+        on install_metrics.date_day = crashes.date_day
+        and install_metrics.package_name = crashes.package_name
+),
```

Comment:
yeah i was considering doing the store_performance stuff in an intermediate model, but it would be a really really short model (which is ok i suppose). open to opinions!

Reply:
I think the shorter intermediate model would be best to then flow down into the final model. If we can break this into an intermediate that would be great!